executor/src/witgen/jit/processor.rs

```diff
@@ -293,6 +320,78 @@ impl<'a, T: FieldElement, FixedEval: FixedEvaluator<T>> Processor<'a, T, FixedEv
     }
 }
 
+/// Returns all variables occurring in the identity.
+/// This also resolves intermediate polynomials.
+fn variables_in_identity<'a, T: FieldElement>(
+    fixed_data: &'a FixedData<'a, T>,
+    identity: &'a Identity<T>,
+    row_offset: i32,
+) -> impl Iterator<Item = Variable> + 'a {
+    identity
+        .children()
+        .flat_map(move |e| variables_in_expression(fixed_data, e, row_offset))
+}
+
+/// Returns all variables occurring in the expression.
+/// This also resolves intermediate polynomials.
+fn variables_in_expression<'a, T: FieldElement>(
+    fixed_data: &'a FixedData<'a, T>,
+    expression: &'a Expression<T>,
+    row_offset: i32,
+) -> impl Iterator<Item = Variable> + 'a {
+    expression
+        .all_children()
+        .flat_map(move |e| -> Box<dyn Iterator<Item = Variable> + 'a> {
+            match e {
+                Expression::Reference(r) => match r.poly_id.ptype {
+                    PolynomialType::Constant => Box::new(std::iter::empty()),
+                    PolynomialType::Committed => {
+                        Box::new(std::iter::once(Variable::from_reference(r, row_offset)))
+                    }
+                    PolynomialType::Intermediate => {
+                        let definition = &fixed_data.intermediate_definitions[&r.to_thin()];
+                        Box::new(variables_in_expression(fixed_data, definition, row_offset))
+                    }
+                },
+                Expression::PublicReference(_) | Expression::Challenge(_) => {
+                    // TODO we need to introduce a variable type for those.
+                    Box::new(std::iter::empty())
+                }
+                _ => Box::new(std::iter::empty()),
+            }
+        })
+}
+
+/// Sorts identities by row and then by ID.
```

Comment:
Is there an easier way?

Reply:
We could implement it for `Identity` - any reason not to compare identities by ID?